src/App.tsx

```diff
@@ -1,25 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import Pagination from './components/Pagination/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPages] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const handleChangePerPages = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setPerPages(+event.target.value);
+    setCurrentPage(1);
+  };
+
+  const startElement = (currentPage - 1) * perPage + 1;
+  const endElement = Math.min(currentPage * perPage, 42);
```

Comment:
The calculation of 'startElement' and 'endElement' is tightly coupled with the pagination logic. Consider encapsulating this logic into a separate function or a custom hook to improve readability and maintainability.